content/docs/concepts/oakestra-extensions/addons.md

```diff
@@ -0,0 +1,87 @@
+
+Addons system is composed of two subsystems:
+- **Addons Marketplace**: Consists of Marketplace Manager and MongoDB client.
+- **Addons Engine**: Consists of Addons Manager, Addons Monitor, and MongoDB Client.
+
+#### Addons Marketplace:
+
+![](marketplace_light.png)
+
+The marketplace allows developers to publish their add-ons for Oakestra. To create
+an addon, users must package their code and convert it into an OCI-compliant image that is
+publicly accessible. Currently, only docker images are supported, but it should be possible to
+use any OCI-compliant image in the future. The Addons Marketplace subsystem offers API
+endpoints for developers to publish their addons to Oakestra. Upon receiving a request for
+registering an Addon, the marketplace asynchronously checks if the image is valid. Once
+it’s verified, the addon’s state in the database changes from `under_review` to `approved`. If
+invalid, the status becomes `failed_verification`.
+
+#### Addons Engine:
+
+![](engine_light.png)
+
+The addons engine is composed of two main components:
+
+- **Addons manager**: handles user requests for installing and uninstalling addons.
+- **Addons Monitor**: is responsible for continually monitoring and managing the state of
+addons. It uses a pull-based mechanism to regularly fetch addon-related tasks from the Addons Manager. It also interacts with the underlying container management engine,
+such as Docker, to manage the life cycle of addons. This includes handling execution,
+monitoring, and reporting of addon statuses to ensure that addons are running as
+expected and reporting any failures to the addons manager. The Addons Monitor
+performs the following checks:
+  - **Install check**: Retrieves all addons from the Addons Manager that need installing.
+  - **Uninstall check**: Retrieves all addons from the Addons Manager that need unin-
+  stalling.
+  - **Cleanup check**: Communicates with underlying containers engines to retrieve
+  all running addons and validates that these addons exist in Addons Manager.
+  The purpose of this is to check if an addon was deleted from the database in the
+  Addons Manager.
+
+
+## Addon Lifecycle Management
+
+The lifecycle of an addon—from creation to deployment—is managed by the **Addons Engine** and the **Addons Marketplace**. This design allows for **hot-swapping**, where new features can be added or swapped without disrupting core services.
+
+When installing an addon, the Addons Engine checks if a similar core component is running. If thats the case, the core component is stopped and the addon is run in its place. In this scenario the addon inherits the port mappings and network specifications of the core component. If no core component exists than its an extension and doesn't replace any other component.
```

Comment:
Please add link to the "Manual" docs in https://github.com/nitindermohan/oak-new-docs/pull/19 at the bottom of the page so there is clear pathway to learn more.